iceoryx_posh/source/runtime/posh_runtime_impl.cpp

```diff
@@ -0,0 +1,579 @@
+    return cxx::error<IpcMessageErrorType>(IpcMessageErrorType::REQUEST_PUBLISHER_WRONG_IPC_MESSAGE_RESPONSE);
+}
+
+SubscriberPortUserType::MemberType_t*
+PoshRuntimeImpl::getMiddlewareSubscriber(const capro::ServiceDescription& service,
+                                         const popo::SubscriberOptions& subscriberOptions,
+                                         const PortConfigInfo& portConfigInfo) noexcept
+{
+    constexpr uint64_t MAX_QUEUE_CAPACITY = SubscriberPortUserType::MemberType_t::ChunkQueueData_t::MAX_CAPACITY;
+
+    auto options = subscriberOptions;
+    if (options.queueCapacity > MAX_QUEUE_CAPACITY)
+    {
+        LogWarn() << "Requested queue capacity " << options.queueCapacity
+                  << " exceeds the maximum possible one for this subscriber"
+                  << ", limiting from " << subscriberOptions.queueCapacity << " to " << MAX_QUEUE_CAPACITY;
+        options.queueCapacity = MAX_QUEUE_CAPACITY;
+    }
+    else if (0U == options.queueCapacity)
+    {
+        LogWarn() << "Requested queue capacity of 0 doesn't make sense as no data would be received,"
+                  << " the capacity is set to 1";
+        options.queueCapacity = 1U;
+    }
+
+    if (options.nodeName.empty())
+    {
+        options.nodeName = m_appName;
+    }
+
+    IpcMessage sendBuffer;
+    sendBuffer << IpcMessageTypeToString(IpcMessageType::CREATE_SUBSCRIBER) << m_appName
+               << static_cast<cxx::Serialization>(service).toString() << cxx::convert::toString(options.historyRequest)
+               << cxx::convert::toString(options.queueCapacity) << options.nodeName
+               << cxx::convert::toString(options.subscribeOnCreate)
+               << cxx::convert::toString(static_cast<uint8_t>(options.queueFullPolicy))
+               << static_cast<cxx::Serialization>(portConfigInfo).toString();
+
+    auto maybeSubscriber = requestSubscriberFromRoudi(sendBuffer);
+
+    if (maybeSubscriber.has_error())
+    {
+        switch (maybeSubscriber.get_error())
+        {
+        case IpcMessageErrorType::SUBSCRIBER_LIST_FULL:
+            LogWarn() << "Service '" << service.operator cxx::Serialization().toString()
+                      << "' could not be created since we are out of memory for subscribers.";
+            errorHandler(Error::kPOSH__RUNTIME_ROUDI_SUBSCRIBER_LIST_FULL, nullptr, iox::ErrorLevel::SEVERE);
+            break;
+        case IpcMessageErrorType::REQUEST_SUBSCRIBER_WRONG_IPC_MESSAGE_RESPONSE:
+            LogWarn() << "Service '" << service.operator cxx::Serialization().toString()
+                      << "' could not be created. Request subscriber got wrong IPC channel response.";
+            errorHandler(Error::kPOSH__RUNTIME_ROUDI_REQUEST_SUBSCRIBER_WRONG_IPC_MESSAGE_RESPONSE,
+                         nullptr,
+                         iox::ErrorLevel::SEVERE);
+            break;
+        default:
+            LogWarn() << "Undefined behavior occurred while creating service '"
```

Comment:
Maybe not the best message to the user. In iceoryx we generally avoid undefined behaviour.

Reply:
How about "Unknown error"?